patroni/dcs/__init__.py

```diff
@@ -1043,32 +1402,51 @@ def get_cluster(self, force: bool = False) -> Cluster:
 
     @property
     def cluster(self) -> Optional[Cluster]:
+        """Cached DCS cluster information that has not yet expired."""
         with self._cluster_thread_lock:
             return self._cluster if self._cluster_valid_till > time.time() else None
 
     def reset_cluster(self) -> None:
+        """Clear cached state of DCS."""
         with self._cluster_thread_lock:
             self._cluster = None
             self._cluster_valid_till = 0
 
     @abc.abstractmethod
     def _write_leader_optime(self, last_lsn: str) -> bool:
-        """write current WAL LSN into `/optime/leader` key in DCS
+        """Write current WAL LSN into ``/optime/leader`` key in DCS.
 
-        :param last_lsn: absolute WAL LSN in bytes
-        :returns: `!True` on success."""
+        :param last_lsn: absolute WAL LSN in bytes.
+
+        :returns: ``True`` if successfully committed to DCS.
+        """
 
     def write_leader_optime(self, last_lsn: int) -> None:
+        """Write value for WAL LSN to ``optime/leader`` key in DCS.
```

Comment:
So perhaps a note to say this is only used on older Patroni versions?

(When I saw that before I did wonder why you need to keep that around. Is it just for clusters upgrading from old versions?)

Reply:
Actually, this method is still called from ha.py on demote or on Patroni shutdown (which also demotes postgres).
It was kept in order to avoid calling write_status() directly, because write_status() requires dict object to be passed and I wanted to abstract it away.

But In any case it first calls write_status() method and only the last one decides whether it should also update /optime/leader

Reply:
Does this addition make sense? https://github.com/zalando/patroni/pull/2777/files#r1279211995

Reply:
Thank you!